src/data/nat/log.lean

```diff
@@ -24,6 +25,34 @@ such that `b^k ≤ n`, so if `b^k = n`, it returns exactly `k`. -/
     log (n / b) + 1
   else 0
 
+lemma log_eq_zero {b n : ℕ} (hnb : n < b ∨ b ≤ 1) : log b n = 0 :=
+begin
+  rw [or_iff_not_and_not, not_lt, not_le] at hnb,
+  rw [log, ←ite_not, if_pos hnb],
+end
+
+lemma log_eq_zero_of_lt {b n : ℕ} (hn : n < b) : log b n = 0 :=
+log_eq_zero $ or.inl hn
+
+lemma log_eq_zero_of_le {b n : ℕ} (hb : b ≤ 1) : log b n = 0 :=
+log_eq_zero $ or.inr hb
+
+lemma log_zero_eq_zero {b : ℕ} : log b 0 = 0 :=
+by { cases b; refl }
+
+lemma log_one_eq_zero {b : ℕ} : log b 1 = 0 :=
+begin
+  by_cases h : b ≤ 1,
+  { exact log_eq_zero_of_le h },
+  { exact log_eq_zero_of_lt (not_le.mp h) },
+end
```

Comment:
Could also be done as:
```suggestion
if h : b ≤ 1 then 
  log_eq_zero_of_le h 
else 
  log_eq_zero_of_lt (not_le.mp h)
```
but this is entirely up to you.

Reply:
I actually prefer the aesthetic of the existing proof, however the proposal seems more intuitive so... thanks!